docs/source/examples/cim.rst

```diff
@@ -0,0 +1,260 @@
+training process in which the agent manager interacts with the environment for collecting experiences and updating
+policies.
+
+.. code-block::python
+
+    env = Env(config.env.scenario, config.env.topology, durations=config.env.durations)
+    agent_id_list = [str(agent_id) for agent_id in env.agent_idx_list]
+    state_shaper = CIMStateShaper(**config.state_shaping)
+    action_shaper = CIMActionShaper(action_space=list(np.linspace(-1.0, 1.0, config.agents.algorithm.num_actions)))
+    if config.experience_shaping.type == "truncated":
+        experience_shaper = TruncatedExperienceShaper(**config.experience_shaping.truncated)
+    else:
+        experience_shaper = KStepExperienceShaper(reward_func=lambda mt: 1-mt["container_shortage"]/mt["order_requirements"],
+                                                  **config.experience_shaping.k_step)
+
+    exploration_config = {"epsilon_range_dict": {"_all_": config.exploration.epsilon_range},
+                          "split_point_dict": {"_all_": config.exploration.split_point},
+                          "with_cache": config.exploration.with_cache
+                          }
+    explorer = TwoPhaseLinearExplorer(agent_id_list, config.general.total_training_episodes, **exploration_config)
+
+    agent_manager = DQNAgentManager(name="cim_learner",
+                                    mode=AgentMode.TRAIN_INFERENCE,
+                                    agent_id_list=agent_id_list,
+                                    state_shaper=state_shaper,
+                                    action_shaper=action_shaper,
+                                    experience_shaper=experience_shaper,
+                                    explorer=explorer)
+
+    actor = SimpleActor(env=env, inference_agents=agent_manager)
+    learner = SimpleLearner(trainable_agents=agent_manager, actor=actor,
+                            logger=Logger("single_host_cim_learner", auto_timestamp=False))
+
+    learner.train(total_episodes=config.general.total_training_episodes)
+
+
+Main Loop with Actor and Learner (Distributed / Multi-process)
+--------------------------------------------------------------
+
+We demonstrate a single-learner and multi-actor topology where the learner drives the program by telling remote actors
+to perform roll-out tasks and using the results they sent back to improve the policies. The workflow usually involves
+launching a learner process and an actor process separately. Because training occurs on the learner side and inference
+occurs on the actor side, we need to create appropriate agent managers on both sides.
+
+On the actor side, the agent manager must be equipped with all shapers as well as an explorer. Thus, The code for
+creating an environment and an agent manager on the actor side is similar to that for the single-host version,
+except that it is necessary to set the AgentMode to INFERENCE. As in the single-process version, the environment and the

```

Comment:
How about applying the format like ``AgentMode.INFERENCE`` (\`\`AgentMode.INFERENCE\`\`) to key words?